oci/spec_opts_linux.go

```diff
@@ -26,22 +26,62 @@ import (
 
 	"github.com/containerd/containerd/containers"
 	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
 	"golang.org/x/sys/unix"
 )
 
 // WithHostDevices adds all the hosts device nodes to the container's spec
 func WithHostDevices(_ context.Context, _ Client, _ *containers.Container, s *Spec) error {
 	setLinux(s)
 
-	devs, err := getDevices("/dev")
+	devs, err := getDevices("/dev", "")
 	if err != nil {
 		return err
 	}
 	s.Linux.Devices = append(s.Linux.Devices, devs...)
 	return nil
 }
 
-func getDevices(path string) ([]specs.LinuxDevice, error) {
+// WithDevices recursively adds devices from the passed in path and associated cgroup rules for that device.
+// If devicePath is a dir it traverses the dir to add all devices in that dir.
```

Comment:
What is the expected behavior when it is a symlink?

Reply:
I expect this to error out, currently since I didn't fix that.

Reply:
So yes it would end up getting `ErrNotADevice`.

I looked at implementing support for symlinks at this level, and it isn't completely straight forward, because we end up mounting the device node into the container instead of the symlink+device... and maybe that doesn't matter. Just messing around with it it felt a little off.
In any case I added a 2nd commit for this case.

Reply:
With `ctr run --privileged` this adds all devices from /dev now AND those same devices again at any symlinked paths, but as real device nodes and not symlinks.

Reply:
I removed the 2nd commit. It is causing test failures the implementation feels wrong here.
Probably the caller will need to handle symlinked paths somehow to create them as symlinks in the rootfs.

Reply:
It looks like the existing behavior (line 117) is to ignore ErrNotADevice so it won't error out on symlink?